### DISCUSSION TITLE
How to load textures from files?

### DISCUSSION THREAD

**Discussion by johnrso** [haosulab/SAPIEN#94] [General]
it is possible to load textures from files (for i.e. domain randomization)? if so, could someone provide an example/point me in a direction?

thanks!

**Comment by fbxiang** [comment]
To load a texture, you can use `renderer.create_texture_from_file` and `material.set_xxx_texture` or `material.set_xxx_texture_from_file` directly. See
https://github.com/haosulab/SAPIEN/blob/02287b2a2ebdce5f38f637577f6debe4fbb757ce/python/pysapien_content.hpp#L2189
https://github.com/haosulab/SAPIEN/blob/02287b2a2ebdce5f38f637577f6debe4fbb757ce/python/pysapien_content.hpp#L543-L561

You then need to assign the material to actors/articulations when building them.
If you want to randomize texture for articulations loaded from URDF files, while you can do it in SAPIEN, I think a easier way is to parse and generate new URDF files.

**Comment by johnrso** [comment]
Thank you! and just to double check, what is the format for texture files?

**Reply by fbxiang** [reply]
You can use your favorite image format (JPG, PNG, etc.)
